Project Proposal - SSF consensus research (#306)
eth-protocol-fellows/cohort-five: projects/ssf-consensus-research.md

> # Project Proposal

Reviewer: This should be name of the project
```suggestion
# SSF Research
```

> A novel consensus mechanism to acheive single slot finality with dynamic availability

Reviewer: ```suggestion
A novel consensus mechanism to achieve single slot finality with dynamic availability
```

## Motivation

Ethereum currently uses the Gasper protocol as its consensus mechanism. It is a complicated and intervowen mechanism between LMD-GHOST that provides availabilty and Casper-FFG which is responsible for finality. This creates several issues like reorg related attacks and longer finality times. Hence, there is an active effort towards Single Slot Finality. 

BFT mechanisms like Tendermint already provide single slot finality but using such protocols is an issue because they can halt if a supermajority of nodes are not online. Ethereum prioritises dynamic availabilty at all costs which makes existing designs like Tendermint infeasible for Ethereum. 

> There has been research into this feild and the development of a simple Single Slot Finality protocol by Francesco D’Amato and Luca Zanolini. The core idea behind there approach is to add another voting phase in the slot for fast finality. The main problem behind this design is the practical limitations of signature aggregation for millions of validators. 

Reviewer 2: ```suggestion
There has been research into this feild and the development of a simple Single Slot Finality protocol by Francesco D’Amato and Luca Zanolini. The core idea behind their approach is to add another voting phase in the slot for fast finality. The main problem behind this design is the practical limitations of signature aggregation for millions of validators. 
```

### Why work on a novel mecahnism when there are several SSF protocols? 
- LMD-GHOST like protocols would require committee based architectures that don't work well for dynamic availability
- there are other problems that arise in GHOST like designs that may not be present in a novel protocol
- prevents tunnel view designs and provides a broad set of protocols to evaluate from

## Specification

The goal of the project would be to come up with a novel design and possibly publish it as an academic research paper. To achieve this, I must first continue going through existing literature and familiarise with research already done in this feild. After gaining a deep understanding of consensus protocols, I will begin drawing out new designs that can provide finality with economic security within a single slot while also preserving dynamic availability. 

> There has been little research on dynmically available protocols so there must also be work done in that direction that can complipent a novel protocol. Mostly the work done during the fellowship will comprise of reading academic research on all related topics and trying to come with new architectures. Ocasional discussions and brainstorming sessions with subject experts can be very helpful.

Reviewer 2: ```suggestion
There has been little research on dynmically available protocols so there must also be work done in that direction that can complement a novel protocol. Mostly the work done during the fellowship will comprise of reading academic research on all related topics and trying to come with new architectures. Occasional discussions and brainstorming sessions with subject experts can be very helpful.
```
